core/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -297,6 +298,13 @@ static void init(String[] args) throws Throwable {
         }
     }
 
+    @SuppressForbidden(reason = "Sets system properties passed as CLI parameters")
+    private static void elasticsearchSettings(Map<String, String> esSettings) {
+        for (Map.Entry<String, String> esSetting : esSettings.entrySet()) {
+            System.setProperty(esSetting.getKey(), esSetting.getValue());
```

Comment:
I do know that at least the logging API used to modify the system property, although I wholeheartedly agree with your point.

https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/common/logging/Loggers.java#L41